velocity-engine-core/src/main/java/org/apache/velocity/util/ArrayIterator.java

```diff
@@ -42,7 +37,8 @@
  * @author <a href="mailto:geirm@apache.org">Geir Magnusson Jr.</a>
  * @version $Id$
  */
-public class ArrayIterator implements Iterator
+public class ArrayIterator
+    implements Iterator<Object>
```

Comment:
Great idea.. but the class who is calling uses Object... maybe in the next commit until understand which type will be the right one.